host/class/uvc/usb_host_uvc/host_test/main/test_fixtures.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * SPDX-FileCopyrightText: 2025 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <catch2/catch_test_macros.hpp>
+#include "mock_add_usb_device.h"
+#include "test_fixtures.hpp"
+
+extern "C" {
+#include "Mockusb_host.h"
+}
+
+esp_err_t test_uvc_host_install(const uvc_host_driver_config_t *driver_config)
+{
+    // Allocation of CTRL transfer (common for all devices)
+    usb_host_transfer_alloc_ExpectAndReturn(64, 0, nullptr, ESP_OK);
+    usb_host_transfer_alloc_IgnoreArg_transfer();
+    // usb_host_transfer_alloc_ExpectAnyArgsAndReturn(ESP_OK);
+    usb_host_transfer_alloc_AddCallback(usb_host_transfer_alloc_mock_callback);
+
+    // Client registration
+    usb_host_client_register_ExpectAnyArgsAndReturn(ESP_OK);
+    usb_host_client_register_AddCallback(usb_host_client_register_mock_callback);
+
+    // Client processing
+    usb_host_client_handle_events_ExpectAnyArgsAndReturn(ESP_OK);
+    usb_host_client_handle_events_AddCallback(usb_host_client_handle_events_mock_callback);
+
+    // Call the real function
+    return uvc_host_install(driver_config);
+}
+
+esp_err_t test_uvc_host_uninstall(void)
+{
+    usb_host_client_unblock_ExpectAnyArgsAndReturn(ESP_OK);
+    usb_host_client_unblock_AddCallback(usb_host_client_unblock_mock_callback);
+
+    usb_host_client_deregister_ExpectAnyArgsAndReturn(ESP_OK);
+    usb_host_client_deregister_AddCallback(usb_host_client_deregister_mock_callback);
+
+    // Free the common CTRL transfer
+    usb_host_transfer_free_ExpectAnyArgsAndReturn(ESP_OK);
+    usb_host_transfer_free_AddCallback(usb_host_transfer_free_mock_callback);
+
+    // Call the real function
+    return uvc_host_uninstall();
+}
+
+esp_err_t usb_host_transfer_submit_control_success_mock_callback(usb_host_client_handle_t client_hdl, usb_transfer_t *transfer, int call_count)
+{
+    transfer->status = USB_TRANSFER_STATUS_COMPLETED;
+    transfer->actual_num_bytes = transfer->num_bytes;
+    transfer->callback(transfer);
+    return ESP_OK;
+}
+
+esp_err_t test_uvc_host_stream_open(const uvc_host_stream_config_t *stream_config, int timeout, uvc_host_stream_hdl_t *stream_hdl_ret, bool is_isoc)
+{
+    /* **Setup callbacks** */
+
+    // This will setup or mocking framework
+    usb_host_device_open_Stub(usb_host_device_open_mock_callback);
+    usb_host_get_device_descriptor_Stub(usb_host_get_device_descriptor_mock_callback);
+    usb_host_device_close_Stub(usb_host_device_close_mock_callback);
+    usb_host_get_active_config_descriptor_Stub(usb_host_get_active_config_descriptor_mock_callback);
+    usb_host_device_addr_list_fill_Stub(usb_host_device_addr_list_fill_mock_callback);
```

Comment:
FYI @peter-marcisovsky I found the `_Stub` CMock functions.

They work the same as `_AddCallback` but CMock stops counting the calls and stops checking the arguments.

So they seem to be suitable for setting up a mock framework, like we do.

Here I set Stubs for usb_host lib functions that I don't care about. I only need the mock_callbacks to be called

Reply:
This PR can be merged after the mock in esp-idf is updated